conda_devenv/devenv.py

```diff
@@ -2,24 +2,61 @@
 
 import argparse
 import os
+import re
 import sys
 
 import six
 
 
+def preprocess_selector_in_line(line):
+    pattern = r"(.*?)#\s+(\[)(.*)(\])(.*)"
```

Comment:
We can cache this for performance and also simplify the grouping a bit (we don't need all groups):

```python
_selector_pattern = re.compile(r".*?#\s*\[(.*)\].*")

def preprocess_selector_in_line(line):
    x = _selector_pattern .search(line)
    ...
    expr = x.group(1).strip()
```

We also don't need to require a space betwen `#` and `[`, because if somebody forgets to add it by accident we would silently include the line regardless of the expression. 👍

Reply:
Good points!